manim/utils/tex.py

```diff
@@ -1,298 +1,189 @@
-        "ctex_packages": [["ctex", ["UTF8"]]],
-        "common_preamble_text": r"\linespread{1}" "\n",
-        "tex_preamble_text": r"\DisableLigatures{encoding = *, family = *}" "\n",
-        "ctex_preamble_text": "",
-        "document_prefix": "",
-        "document_suffix": "",
-    }
-
-    def __init__(self, **kwargs):
-        digest_config(self, kwargs)
-        self.rebuild_cache()
-
-    def rebuild_cache(self):
-        """For faster access, the LaTeX template's code is cached.
-        If the base file is modified, the cache needs to be rebuilt."""
-        tpl = self.generate_tex_command(
-            "documentclass",
-            required_params=[self.documentclass[0]],
-            optional_params=self.documentclass[1],
+        self.output_format = (
+            output_format
+            if output_format is not None
+            else TexTemplate.default_output_format
+        )
+        self.documentclass = (
+            documentclass
+            if documentclass is not None
+            else TexTemplate.default_documentclass
+        )
+        self.preamble = (
+            preamble if preamble is not None else TexTemplate.default_preamble
+        )
+        self.placeholder_text = (
+            placeholder_text
+            if placeholder_text is not None
+            else TexTemplate.default_placeholder_text
+        )
+        self.post_doc_commands = (
+            post_doc_commands
+            if post_doc_commands is not None
+            else TexTemplate.default_post_doc_commands
+        )
+        self.rebuild()
+
+    def rebuild(self):
+        self.body = (
+            self.documentclass
+            + "\n"
+            + self.preamble
+            + "\n"
+            + r"\begin{document}"
+            + "\n"
+            + self.post_doc_commands
+            + "\n"
+            + self.placeholder_text
+            + "\n"
+            + r"\end{document}"
+            + "\n"
         )
-        for pkg in self.common_packages:
-            tpl += self.generate_usepackage(pkg)
-
-        if self.use_ctex:
-            for pkg in self.ctex_packages:
-                tpl += self.generate_usepackage(pkg)
-        else:
-            for pkg in self.tex_packages:
-                tpl += self.generate_usepackage(pkg)
-
-        tpl += self.common_preamble_text
-        if self.use_ctex:
-            tpl += self.ctex_preamble_text
-        else:
-            tpl += self.tex_preamble_text
-
-        tpl += "\n" r"\begin{document}" "\n"
-        tpl += f"\n{self.text_to_replace}\n"
-        tpl += "\n" r"\end{document}"
-
-        self.body = tpl
 
-    def prepend_package(self, pkg):
-        """Adds a new package (or several new packages)
-        before all other packages. Sometimes, the order of
-        the ``\\usepackage`` directives is relevant.
+    def prepend_to_preamble(self, txt):
+        """Adds txt to the TeX template preamble just after the \documentclass
 
         Parameters
         ----------
```

Comment:
```suggestion

```

Not needed anymore if we have `add_to_preample` with parameter for prepending/appending.